pkg/destroy/openstack/openstack.go

```diff
@@ -522,6 +495,145 @@ func deleteRouters(opts *clientconfig.ClientOpts, filter Filter, logger logrus.F
 	return len(allRouters) == 0, nil
 }
 
+func removeRouterInterfaces(client *gophercloud.ServiceClient, customSubnets []string, routerID string, logger logrus.FieldLogger) (bool, error){
+	// Get router interface ports
+	portListOpts := ports.ListOpts{
+		DeviceID: routerID,
+	}
+	allPagesPort, err := ports.List(client, portListOpts).AllPages()
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+	allPorts, err := ports.ExtractPorts(allPagesPort)
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+
+	var customInterfaces []ports.Port
+	// map to keep track of whethere interface for subnet was already removed
+	removedSubnets := make(map[string]bool)
+	for _, port := range allPorts {
+		for _, IP := range port.FixedIPs {
+			if lookupOpenStackSubnet(customSubnets, IP.SubnetID, logger) {
```

Comment:
perhaps we can rename it, as all are openstack subnet anyway. Though I don't have a good name for it either. Perhaps this could be internal funcion, and then have a wrapper calling it isPrimarySubnet, and/or isKuryrNetwork for the extra ones